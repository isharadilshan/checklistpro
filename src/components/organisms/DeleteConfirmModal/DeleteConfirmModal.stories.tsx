import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import DeleteConfirmationModal from './';

export default {
  title: 'components/organisms/DeleteConfirmationModal',
  component: DeleteConfirmationModal,
} as ComponentMeta<typeof DeleteConfirmationModal>;

export const Basic: ComponentStory<typeof DeleteConfirmationModal> = () => {
  const args = {
    modalVisible: true,
    closeModal: () => {},
    onDeleteConfirm: () => {},
  };
  return (
    <NativeBaseProvider>
      <DeleteConfirmationModal {...args} />
    </NativeBaseProvider>
  );
};
