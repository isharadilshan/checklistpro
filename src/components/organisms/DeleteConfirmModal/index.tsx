import React, {useRef} from 'react';
import {Button, Modal, Text} from 'native-base';
import createStyle from './styles';

type DeleteConfirmModalProps = {
  modalVisible: boolean;
  closeModal: () => void;
  onDeleteConfirm: () => void;
};

const DeleteConfirmModal: React.FC<DeleteConfirmModalProps> = ({
  modalVisible,
  closeModal,
  onDeleteConfirm,
}) => {
  const initialRef = useRef(null);
  const finalRef = useRef(null);
  const styles = createStyle();

  return (
    <Modal
      size={'xl'}
      avoidKeyboard
      isOpen={modalVisible}
      onClose={closeModal}
      initialFocusRef={initialRef}
      finalFocusRef={finalRef}
    >
      <Modal.Content style={styles.modalContent}>
        <Modal.CloseButton />
        <Modal.Header
          style={styles.modalHeader}
          _text={{color: 'blueGray.200'}}
        >
          Delete Expense
        </Modal.Header>
        <Modal.Body>
          <Text color={'blueGray.200'}>Are you sure to delete modal</Text>
        </Modal.Body>
        <Modal.Footer style={styles.modalFooter}>
          <Button.Group>
            <Button variant="ghost" colorScheme="blueGray" onPress={closeModal}>
              Cancel
            </Button>
            <Button onPress={onDeleteConfirm}>Confirm</Button>
          </Button.Group>
        </Modal.Footer>
      </Modal.Content>
    </Modal>
  );
};

export default DeleteConfirmModal;
