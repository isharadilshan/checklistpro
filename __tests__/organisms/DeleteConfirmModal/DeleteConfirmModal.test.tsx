import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import DeleteConfirmModal from '../../../src/components/organisms/DeleteConfirmModal';

configure({adapter: new Adapter()});

const props = {
  modalVisible: true,
  closeModal: jest.fn(),
  onDeleteConfirm: jest.fn(),
};

describe('when rendering', () => {
  const wrapper = shallow(<DeleteConfirmModal {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
