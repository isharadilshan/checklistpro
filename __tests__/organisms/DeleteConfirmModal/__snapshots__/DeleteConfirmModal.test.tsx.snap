// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<Memo(ForwardRef(Modal))
  avoidKeyboard={true}
  finalFocusRef={
    {
      "current": null,
    }
  }
  initialFocusRef={
    {
      "current": null,
    }
  }
  isOpen={true}
  onClose={[MockFunction]}
  size="xl"
>
  <Memo(ForwardRef(ModalContent))
    style={
      {
        "backgroundColor": "#141E30",
      }
    }
  >
    <Memo(ForwardRef(ModalCloseButton)) />
    <Memo(ForwardRef(ModalHeader))
      _text={
        {
          "color": "blueGray.200",
        }
      }
      style={
        {
          "backgroundColor": "#141E30",
          "borderBottomWidth": 0,
        }
      }
    >
      Delete Expense
    </Memo(ForwardRef(ModalHeader))>
    <Memo(ForwardRef(ModalBody))>
      <Memo(ForwardRef(Text))
        color="blueGray.200"
      >
        Are you sure to delete modal
      </Memo(ForwardRef(Text))>
    </Memo(ForwardRef(ModalBody))>
    <Memo(ForwardRef(ModalFooter))
      style={
        {
          "backgroundColor": "#141E30",
          "borderTopWidth": 0,
        }
      }
    >
      <Memo(ForwardRef)>
        <Memo(ForwardRef(Button))
          colorScheme="blueGray"
          onPress={[MockFunction]}
          variant="ghost"
        >
          Cancel
        </Memo(ForwardRef(Button))>
        <Memo(ForwardRef(Button))
          onPress={[MockFunction]}
        >
          Confirm
        </Memo(ForwardRef(Button))>
      </Memo(ForwardRef)>
    </Memo(ForwardRef(ModalFooter))>
  </Memo(ForwardRef(ModalContent))>
</Memo(ForwardRef(Modal))>
`;
